[CS2103T-W13-1] Module Manager

geshuming opened the pull request:
@alexwxh
@bitterg0d
@CowSaysBaa
@geshuming
@Kzrthikz

geshuming committed:
Update Badges on Readme

Kzrthikz committed:
Update Contact Us Page (#11)

* null commit

* null commit

* contact us updated

* updated contact-us

bitterg0d committed:
Update README.adoc (#13)

* updated README.adoc and Ui.png

* updated README.adoc

* Update README.adoc

CowSaysBaa committed:
Updated AboutUs.adoc (#10)

* added some images and changed AboutUs.adoc

* updated photos and AboutUs.adoc

* Added all photos and updated AboutUs.adoc

* Updated AboutUs title

* Delete johndoe.adoc

* deleted some unused images

* Merge branch 'master' of https://github.com/AY1920S1-CS2103T-W13-1/main

geshuming committed:
Update developer guide (#9)

* Update Developer Guide

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

le0tan commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to find a module with incomplete keyword (e.g. 'CS2')
```
Can we include the extension where the requested module doesn't exist?

le0tan commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-4.  AddressBook deletes the person
+1.  User requests to find a module with incomplete keyword (e.g. 'CS2')
+2.  ModuleBook shows a list of modules matching the keyword (e.g. 'CS2030, CS2040, CS2100, CS2101, etc.')
+3.  User requests to track a module
```
What if user gives up tracking a module after ModuleBook shows the matching modules?

le0tan commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+1.  User requests to find a module with incomplete keyword (e.g. 'CS2')
+2.  ModuleBook shows a list of modules matching the keyword (e.g. 'CS2030, CS2040, CS2100, CS2101, etc.')
+3.  User requests to track a module
+4.  ModuleBook tracks the module
```
Maybe after it tracks the module, it could indicate the user that the tracking is successful?

tiuweehan commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |busy student |prioritize certain modules (and their deadlines) |manage my time better
+
+|`* *` |organized student |manage files related to a module |add and tag files
```
The benefit and functions for this use case seems to be swapped

LuWenQ commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-* 2a. The list is empty.
+*MSS*
+
+1.  User requests to view the specific tracked module
+2.  ModuleBook displays the module details
+3.  User requests to add a deadline
+4.  ModuleBook adds the deadline
```
Consider adding a "." after every line of MSS?

tiuweehan commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |student |generate a sample module planner |plan my modules
+
+|`*` |hardworking student |access past exam papers |
```
The benefits for a function is subjective and may not be immediately obvious, so it would be safe to include the benefits for all user stories

Seris370 replied:
Would it be better not to omit the benefit of this user story?

tiuweehan commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |busy student |manage deadlines |keep track of my assignments and tasks
+
+|`* * *` |organized student |manage my grades |know roughly how well I'm doing in the module
```
It is not very clear what managing grades mean. It seems similar to calculating GPA

Seris370 replied:
What does it mean by managing grades? Is it the same as calculating GPA?

LuWenQ commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+** 2a2. The module is now tracked
+
 +
+Use case resumes at step 3
```
Consider changing the indentation?

jiayushe commented on docs/DeveloperGuide.adoc:
```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 _{More to be added}_
 
 [appendix]
 == Non Functional Requirements
```
There are many more NFRs that can be included. For example,
- Should work on both 32-bit and 64-bit enviroments
- Should be able to manage up to 200 modules without noticeable sluggishness in performance
- Should work without installation
- Should work without internet access
...

tiuweehan reviewed (commented):
The user stories are comprehensive and covers many potential uses.

alexwxh committed:
update user guide (#12)

* update UG

* update UG according to feedback

* add newline at EOF

CowSaysBaa committed:
Migrate AddressBook -> ModuleBook (#16)

* added some images and changed AboutUs.adoc

* updated photos and AboutUs.adoc

* Added all photos and updated AboutUs.adoc

* Renamed Person to Module

* Removed all unnecessary classes, created stub for AddCommand

* renamed some classes with person to modules
fixed ability to delete entries
base skeleton should be functional

* Fixed checkstyle

* Merge branch 'skeleton-branch' of https://github.com/CowSaysBaa/main

# Conflicts:
#	docs/UserGuide.adoc (resolved)

* Remove tutorial.
Resolves issues related to naming and spacing.

* deleted tutorials

* Resolved issues with naming and spacing

* Resolved whitespacing issue in JsonUtil.java

* Rename address/person -> module

* Rename main package address -> module

* Update build.gradle

* Rename test package address -> module

* Overhaul test package for migration to ModuleBook

* Modify help window constants

* Fix failing checkstyle issues

* Rename address -> module in LogsCenter

* Remove setModule

* Modify Module's identity fields

geshuming committed:
Add archivedModules.json (#28)

* Add archivedModules.json

* Update README.adoc

* Update archivedModules.json

geshuming committed:
Add ArchivedModule and Storage support (#27)

* Implement ArchivedModule and ArchivedModuleList

* Add archivedModules.json

* Modify ModuleBook to aggregate ArchivedModule

* Add support to read ArchivedModules from file

* Revert "Add archiveModules.json"

* Fix equals method in ArchivedModule

* Add tests for JsonArchivedModule

* Extract readArchivedModules into JsonArchivedModuleList

* Rename isSameModule -> isSameArchivedModule

* Rename ModuleBuilder -> ArchivedModuleBuilder

* Add ArchivedModuleTest

* Add ArchivedModuleListTest

* Add JsonArchivedModuleListTest.java

* Update JsonUtilTest

* Revert JsonModuleBookStorageTest

CowSaysBaa committed:
Find command (#31)

* Changed the logic for the find command to search within ArchivedModuleList instead of ModuleList
Added ArchivedNameContainsKeywordsPredicate.java

* Added some fullstops to javadocs.

* Updated ModuleBook to copy ArchiveModuleList as well.

* Renamed: Module -> TrackedModule.

* Added: Module.java interface.

* Type migrated TrackedModule to Module in NameContainsKeywordsPredicate.java

* Deleted ArchivedNameContainsKeywordsPredicate.java.
Made some type migration.

* Added displayedList to ModelManager.java.
Changed UI to display displayedList instead.

* Changed how displayedList works to dislay ArchivedList and TrackedList.
List and Find command should work now.

* Reverted checkstyle.xml.

* Reverted unintentional renames.

* Reverted unintentional renames.

* Fixed checkstyle.

* Fixed checkstyles.

* Fixed typing issues.
Fixed some naming issues.

* Split changeDisplayedList to displayArchivedList and displayTrackedList.
Fixed some styling.

* Fixed naming issue.

CowSaysBaa committed:
Add and delete command (#33)

* added some images and changed AboutUs.adoc

* updated photos and AboutUs.adoc

* Added all photos and updated AboutUs.adoc

* Fixed DeleteCommand.java.
DeleteCommand should now work.

* Changed the fields and accessors within TrackedModule.java.

* Deleted ArchiveStub.
TrackedModule now reference ArchivedModule.
Changed how add command works.

* Require ModuleBook to init with ArchivedModuleList

* Fix JsonAdaptedModule

* Fix checkstyle issues

* Delete SampleDataUtil

* Simplify ModuleBook constructor

* Fix ModuleBook not initializing with archived modules

* Fix Add Command not refreshing displayed list

geshuming committed:
Fix image not showing in github website (#26)

geshuming committed:
Modify JsonArchivedModuleList to read from resource directly (#44)

geshuming committed:
Update ModelClassDiagram and DeveloperGuide (#29)

* Update ModelClassDiagram

* Update ModelClassDiagram for #31

* Update ModelClassDiagram for #31

* Update DeveloperGuide

Kzrthikz committed:
Add deadline feature (#38)

* AddDeadlineCommand and DeleteDeadlineCommand class

* add ArgumentMultimap and ArgumentTokenizer

* add ArgumentMultimap and ArgumentTokenizer

* editing AddDeadlineCommandParser

* edit deadline, pending to integrate into gui

* setModule from ModuleBook

* setModule from ModuleBook

* add deadline to module completed

* edited UserGuide doc

* writing some tests

* refractored TrackedModule and ArchivedModule

* tests for TrackedModule

* cleanup code

* cleanup code

* Fix broken code

geshuming committed:
Implement Module Viewing Feature (#46)

* Updated GUI to include a main panel

* Add View Command

* Update GUI to reflect new module view

* Fix failing tests

* Add Back Command

* Refactor to use Optional for displayedModule

* Add BackCommandTest

* Fix ModelManager.setModuleBook not resetting the filtered lists

* Add ViewCommandTest

* Fix failing checkstyle

* Update javadocs

geshuming committed:
Update StorageClassDiagram and DeveloperGuide (#30)

* Update Storage Diagram

* Update StorageClassDiagram for #33

* Fix arrow showing extension instead of dependency

* Update DeveloperGuide

geshuming committed:
Add extra check to run travis check scripts (#56)

geshuming committed:
Cleanup DG (#52)

* Cleanup DG

* Remove unreferenced images

geshuming committed:
Update UG and DG for Module View Feature (#47)

* Update User Guide

* Update Developer Guide

* Update ModuleViewSequenceDiagram

Kzrthikz committed:
Refractor Deadline  (#57)

* AddDeadlineCommand and DeleteDeadlineCommand class

* add ArgumentMultimap and ArgumentTokenizer

* add ArgumentMultimap and ArgumentTokenizer

* editing AddDeadlineCommandParser

* edit deadline, pending to integrate into gui

* setModule from ModuleBook

* setModule from ModuleBook

* add deadline to module completed

* edited UserGuide doc

* writing some tests

* refractored TrackedModule and ArchivedModule

* tests for TrackedModule

* cleanup code

* cleanup code

* Fix broken code

* split into respective DeadlineCommands, DeadlineCommandParser, Create List for Deadlines

* split into respective DeadlineCommands, DeadlineCommandParser, Create List for Deadlines

* Fix minor changes

* Commit changes

* Refractor Deadline commands

Kzrthikz committed:
Update UG Deadline (#54)

* null commit

* null commit

* Update User Guide

* Update User Guide

geshuming committed:
Cleanup docs to remove references to addressbook (#59)

* Remove header bar

* Remove unnecessary pages

* Update testing page

* Revert "Remove unnecessary pages"

This reverts commit 4a1a83f5cab3845467bde9d8a46c01ea8088a90a.

* Clean up references to AddressBook

CowSaysBaa committed:
Update UG and DG for ListView operations. (#58)

* Updated listview commands on UG.
Updated ArchitectureSequenceDiagram.puml
Updated ArchitectureSequenceDiagram.png

* Renamed DeleteSequenceDiagram to AddSequenceDiagram
Updated AddSequenceDiagram.puml
Updated AddSequenceDiagram.png

* Updated DG to include SearchFiltering extension.
Added FindSequenceDiagram.puml and png.

* Resolved merge conflict in UserGuide.adoc

geshuming committed:
Fix and Update tests for logic package (#55)

* Fix CommandResult and tests

* Fix and test ViewCommandParser

* Update ModuleBookParserTest.

* Update LogicManagerTest

* Add test for getDisplayedModule

jakq commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,94 +149,120 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+The following sequence diagram shows how the module view feature is executed.
 
-image::UndoRedoState2.png[]
+image::ModuleViewSequenceDiagram.png[]
```
Hello! I don't think you can write something like "create and execute"? Unless your method is called "create and execute"? 😺 😺

geshuming replied:
The actual implementation is abstracted out so as to keep the diagram simpler

jakq commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,94 +149,120 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+The following sequence diagram shows how the `ViewCommand` attempts to find the module from the `Model`.
 
-image::UndoRedoState3.png[]
+image::ModuleViewSequenceDiagram_FindModule.png[]
```
Hi! I think after you return Tracked Module List, the activation bar for your Model should end, and start again in the optional path.

alexwxh committed:
add feature Link (#40)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* add feature link

* cleanup

* change parsing to prefix-based

* split linkcommand classes

* minor cleanup

* checkstyle

* fix checkstyle

* checkstyle version revert

* checkstyle

* update userguide link feature

* add link feature to ui

Kzrthikz committed:
Update DG for deadline feature (#65)

* null commit

* null commit

* update DG

* add new line at EOF

* minor updates

* edit addDeadlineSequenceDiagram

* edit size of text in addDeadlineSequenceDiagram

geshuming committed:
Update Module View Documentation (#62)

* Update Module View Documentation

* Add extra details

* Add heading for images

* Update FindModule diagram

* Update FindModule image

geshuming committed:
Revert default fontsize change (#67)

alexwxh committed:
added link feature to DG (#66)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* add feature link

* cleanup

* change parsing to prefix-based

* split linkcommand classes

* minor cleanup

* checkstyle

* fix checkstyle

* checkstyle version revert

* checkstyle

* update userguide link feature

* add link feature to ui

* DG update link

* minor updates to link DG

* DG update

geshuming committed:
Update tests for model package (#68)

* Update Module unit tests

* Update ModuleBookTest

* Update ModelManagerTest

geshuming committed:
Update tests for storage package (#69)

* Update JsonSerializableModuleBookTest

* Update JsonAdaptedModuleTest

* Fix and update JsonModuleBookStorageTest

* Update StorageManagerTest

CowSaysBaa committed:
Refactor Add command. (#70)

* Added Model#findArchivedModue and Module#findTrackedModule.
Added ModelManager#findArchivedModue and ModelManager#findTrackedModule.
Changed AddCommand ViewCommand to use findArchivedModule and findTrackedModule.
Changed AddCommandParser to use Predicate<Module>.

* Added SameModuleCodePredicate.java.
Changed AddCommandParser.java and ViewCommand.java to use SameModulePredicate.java.
Fixed AddCommandParserTest.java

* Fixed ModuleBookParserTest#parseCommand_add()

* Fixed AddCommandTest.java

* Fixed checkstyle

* Changed implementation of ViewCommand to look nicer.

* Removed redundant code.

* Removed unused imports.

CowSaysBaa committed:
Re implement delete command (#71)

* Updated DeleteCommand and DeleteCommandParser to use modulecode instead of index.

* Updated some DeletecommandParserTest test cases.

* Fixed checkstyle.

* Updated DG and UG for delete command.

geshuming committed:
Add more fields to Module View (#72)

* Refactor tests to use builder classes

* Add prerequisite field to ArchivedModule

* Add semesterData to ArchivedModule

* Fix regression

* Add SemesterDetail to ArchivedModule

* Add preclusion to ArchivedModule

* Refactor ModuleViewPanel to make it scrollable

* Cleanup Ui

CowSaysBaa committed:
Filtered search (#74)

* Created package predicate in module. Moved NameContainsKeywordsPredicate.java and SameModuleCodePredicate.java to predicate.

* Added ModuleCodeContainsKeywordsPredicate.java

* Updated FindCommand and FindCommandParser to test a list of predicates.
Updated CommandParserTest.java.

* Fixed test cases in FindCommandTest.java.

* Fixed a test case in ModuleBookParser.

* Changed FindCommand to take in a list of predicates instead of an arraylist of predicates.

* Added DescriptionContainsKeywordsPredicate.java.
Added ability to filter by description.

* Updated UG and DG for filtered search.

* Fixed checkstyle.

* Added PrerequisiteContainsKeywordsPredicate.java.
Added ability to filter search by preculsions.

* Added PreclusionContainsKeywordsPredicate.java
Added ability to filter by preclusion.

* Added SemesterContainsKeywordsPredicate.java.
Added ability to filter modules by semester.

* Updated UG and DG.

* Fixed Checkstyle.

* Fixed test cases.

Kzrthikz committed:
More Deadline Features  (#76)

* AddDeadlineCommand and DeleteDeadlineCommand class

* add ArgumentMultimap and ArgumentTokenizer

* add ArgumentMultimap and ArgumentTokenizer

* editing AddDeadlineCommandParser

* edit deadline, pending to integrate into gui

* setModule from ModuleBook

* setModule from ModuleBook

* add deadline to module completed

* edited UserGuide doc

* writing some tests

* refractored TrackedModule and ArchivedModule

* tests for TrackedModule

* cleanup code

* cleanup code

* Fix broken code

* split into respective DeadlineCommands, DeadlineCommandParser, Create List for Deadlines

* split into respective DeadlineCommands, DeadlineCommandParser, Create List for Deadlines

* Fix minor changes

* Commit changes

* Refractor Deadline commands

* delete deadline feature

* done and in progress deadline feature

* implementing priority command

* clean up existing command features

* Extracted out some common impl

geshuming committed:
Fix datetime stuck in UTC (#78)

geshuming committed:
Update archivedModules.json (#81)

geshuming committed:
Update user guide and Ui.png for 1.3 (#80)

* Update user guide for 1.3

* Update delete command in user guide

* Remove multiple user profiles

alexwxh committed:
Update link feature (#79)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* refactored link command to work in moduleview + add delete link

* update UG

* edit LinkCommand

* remove unused imports

Kzrthikz committed:
Update and fix deadline UI (#87)

* AddDeadlineCommand and DeleteDeadlineCommand class

* add ArgumentMultimap and ArgumentTokenizer

* add ArgumentMultimap and ArgumentTokenizer

* editing AddDeadlineCommandParser

* edit deadline, pending to integrate into gui

* setModule from ModuleBook

* setModule from ModuleBook

* add deadline to module completed

* edited UserGuide doc

* writing some tests

* refractored TrackedModule and ArchivedModule

* tests for TrackedModule

* cleanup code

* cleanup code

* Fix broken code

* split into respective DeadlineCommands, DeadlineCommandParser, Create List for Deadlines

* split into respective DeadlineCommands, DeadlineCommandParser, Create List for Deadlines

* Fix minor changes

* Commit changes

* Refractor Deadline commands

* delete deadline feature

* done and in progress deadline feature

* implementing priority command

* clean up existing command features

* Extracted out some common impl

* Fix broken code. Update UI to ModuleViewPanel.

* fix travis fail

* fix checkstyle

Kzrthikz committed:
Update UG deadline for dry run (#91)

* null commit

* null commit

* update UG for dry run

CowSaysBaa committed:
Added TitleContainsKeywordsPredicate.java (#118)

Added ability to filter by title
Fixed inability to add modules after some search
Fixed UG for search command

geshuming committed:
Add text wrap to result display (#141)

geshuming committed:
Refactor module list and add tracking status (#142)

* Add trackedStatus label

* Update gui

* Overhaul displayedlist to show tracked modules in place of archived modules

* Fix broken tests

* Revert getTrackedModuleByIndex changes

* Add border around trackedStatus pane

geshuming committed:
Update GUi (#144)

* Cleanup CSS and Ui

* Add accordion to module view

* Update home page

alexwxh committed:
fix link-related issues (#143)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* change ui layout for links + fix link adding

* checkstyle

* removed hotfix

* implement edit link command

CowSaysBaa committed:
Added Search UI. (#145)

CowSaysBaa committed:
Fuzzy search (#148)

* Added LevenshteinDistanceChecker.java.
Added support for fuzzy search.
Added test for fuzzy search.

* Added ability to list all archived modules.

* Updated UG.

* Fixed checkstyle.

alexwxh committed:
New link features (#150)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* change ui layout for links + fix link adding

* checkstyle

* removed hotfix

* implement edit link command

* add mark command and button hover text

* checkstyle

* edit link UI

* update userguide

* update UG unmark

geshuming committed:
Add onClick handler to moduleListCell  (#152)

* Add onClick handler to display module when clicked

* Add test for logic setDisplayedModule

* Update UG to reflect new feature

geshuming committed:
Add sample data (#153)

geshuming committed:
Add accordions (#154)

* Add accordions

* Fix Height issues

* Remove link label

geshuming committed:
Attribute code from #33 (#155)

CowSaysBaa committed:
Optimised search (#156)

*  Changed search by description to search by keywords instead of sentence.

* Changed search by title to search keywords instead of a sentence.

* Updated UG.

* Changed seach by description and title to match all keywords instead of matching at least one.

alexwxh committed:
Add error message for invalid url provided (#157)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* change ui layout for links + fix link adding

* checkstyle

* removed hotfix

* implement edit link command

* include invalid url input exception

* Revert "include invalid url input exception"

This reverts commit f790f620ff0714e4a95017e82033c0a5d7611457.

* refactor error handling

* checkstyle

Kzrthikz committed:
Fix issues for Deadline  (#147)

* null commit

* null commit

* fix some issues, format date and time, sort by date, time and priority

* implement deleteAll and doneAll commands

* update UG for deleteAll and doneAll commands

* fix deadline issues

* fix deadline issues

* create unique exception classes

* Fixed all the exception issues

* checkStyle

* Fix storage error

* remove unused exception

Kzrthikz committed:
Edit and undone feature deadline exceptions (#160)

* null commit

* null commit

* Edit and undone error messages

* CheckStyle

* clean up error messages

* remove "Deadline" from deadline drop box in ui

geshuming committed:
Fix images not showing (#158)

geshuming committed:
Update DG (#159)

* Remove reference diagram

* Add new proposed feature: module status

* Add activity diagram to view feature

* Add bolding

* Add code literal

geshuming committed:
Update UG (#161)

* Add new images for glossary

* Move some images

Kzrthikz committed:
Update UG and DG Deadline (#162)

* null commit

* null commit

* update UG

* update UG and DG

* add eof

* edit size of activity diagram

* edit typo

geshuming committed:
Update Ui.png (#163)

geshuming committed:
Update css for website (#164)

* Add text-align:center

CowSaysBaa committed:
Added test cases for predicates. (#166)

* Updated NameContainsKeywordsPredicateTest.java

* Removed unnecessary tests. Fixed some style.

* Moved NameContainsKeywordsPredicateTest.java

* Added SameModuleCodePredicateTest.java

* Added DescriptionCodeContainsKeywordsPredicateTest.java

* Added ModuleCodeContainsKeywordsPredicateTest.java

* Added TitleContainsKeywordsPredicate.java

* Added PreclusionContainsKeywordsPredicateTest.java

* Added PrerequisiteContainsKeywordsPredicateTest.java

* Added equals test.
Fixed checkstyle.

* Changed modulecode assertion.

Kzrthikz committed:
Deadline testcases (#169)

* null commit

* null commit

* tests invalid module, task number & Duplicate task.

geshuming committed:
Fix issue when too many arguments are given to some commands (#171)

geshuming committed:
Update ModuleViewFindModuleActivityDiagram (#170)

geshuming committed:
Add PPP (#167)

* Draft PPP

* Update PPP

* Add link to portfolio

* Add page breaks

CowSaysBaa committed:
Changed how SameModuleCodePredicate works. (#173)

Kzrthikz committed:
Add PPP  (#165)

* null commit

* null commit

* Update PPP

* added eof line

* update PPP

* clean up PPP

CowSaysBaa committed:
Updated UG and DG (#175)

* Updated AddSequenceDiagram

* Updated ArchitectureSequenceDiagram

* Updated Ui component of DG.

* Added FindSequenceDiagram and updated DG.

* Updated UG.

* Added new line to eof.

alexwxh committed:
Test cases for links (#174)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* change ui layout for links + fix link adding

* checkstyle

* removed hotfix

* implement edit link command

* link test cases

* checkstyle + fix errors

geshuming committed:
Update tests (#172)

* Cleanup View Command test

* Cleanup Back Command Test

* Update List Command Test

* checkstyle

* Add test for parser

* Update Module Test

* Add semester detail test

* Remove redundant test

geshuming committed:
Update sample data (#168)

* Update sample data

* Update Ui.png

CowSaysBaa committed:
Add PPP (#88)

* Added PPP.

* Renamed Karthika's PPP

* Updated own PPP.

* Updated AboutUs.

* Added EOF

* Added brackets.

* Changed link to Reposense.

* Added pull request.

Kzrthikz committed:
Deadline testcases (#177)

* null commit

* null commit

* Test case for Deadline Commands

Kzrthikz committed:
Minor edit to PPP (#179)

* null commit

* null commit

* add PR to ppp

* change ppp file name in about us

geshuming committed:
Do not display urls in html to pdf conversion (#181)

* Do not display urls in html to pdf conversion

* checkstyle

CowSaysBaa committed:
Fix screenshot not showing up on website. (#182)

* Fix screenshot not showing up on website.

* Removed figure number reference.

geshuming committed:
Update PPP (#183)

CowSaysBaa committed:
Delete SearchGui.PNG (#184)

Kzrthikz committed:
Edit ppp  (#180)

* null commit

* null commit

* add italise

Kzrthikz committed:
Update UG and DG (#185)

* null commit

* null commit

* add italise

* update ug with pics, dg sequence diagram

* resize image size

* edit AddDeadlineSequenceDiagram

CowSaysBaa committed:
Updated sequence diagram and corrected some grammar. (#187)

geshuming committed:
Update UG and DG for 1.4 (#188)

* Update UIClassDiagram

* Update ModelClassDiagram

* Update UG and DG

* Update StorageClassDiagram

CowSaysBaa committed:
Fixed some minor grammartical mistakes. (#189)

alexwxh committed:
update ugdg&ppp (#190)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* change ui layout for links + fix link adding

* checkstyle

* removed hotfix

* implement edit link command

* edit readme + add icon

* update ugdg ppp

* add portfolio link

* add EOF

* rename ppp

geshuming committed:
Cleanup UG (#191)

geshuming committed:
Update AboutUs (#193)

CowSaysBaa committed:
Added Class diagram to find. (#194)

* Added Class diagram to find.

* Resized find class diagram.

CowSaysBaa committed:
Resized find class diagram. (#195)

alexwxh committed:
Resize images  (#192)

* update UG

* Revert "update UG"

This reverts commit d3c5affab88db6fe69609dff663562291c7b4ff8.

* change ui layout for links + fix link adding

* checkstyle

* removed hotfix

* implement edit link command

* resize images

* resize pt2

* resize pt3

* resize remove delete link ug from ppp

* fix minor errors

* remove whitespace

* phrasing and stuff

* final edit
